decode.go

```diff
@@ -945,9 +945,8 @@ func readTimestamp(r *buffer) (time.Time, error) {
 		return time.Time{}, errorErrorf("invalid type for timestamp %02x", type_)
 	}
 
-	n, err := r.readUint64()
-	rem := n % 1000
```

Comment:
What do you think about converting to signed here and retaining the remainder logic to avoid overflow of the `int64` nanoseconds. It's a corner case since overflow would occur sometime in year 2262, but I think we should handle it.

Reply:
Agreed and pushed. I was under the mistaken impression that time.Time was just int64 nanoseconds internally, but I looked it up and it's got  a bigger range so best not to lose any accuracy.